app/workers/audio_callback_worker.rb

```diff
@@ -9,6 +9,17 @@ class UnknownAudioTypeError < StandardError; end
 
   def perform(_sqs_msg, job)
     audio = AudioFile.find(job['id'])
+    update_audio(audio, job)
+    story = audio.story
+    story.with_lock do
```

Comment:
And you confirmed that `with_lock` reloads the model?

Reply:
`lock!` reloads the record, per [docs](http://api.rubyonrails.org/classes/ActiveRecord/Locking/Pessimistic.html), and `with_lock` uses `lock!` (from lurker gigi)

Reply:
yes, I looked up the code, it essentially does `model.lock!; model.reload; yield model` inside a transaction.